geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/DeltaDUnitTest.java

```diff
@@ -0,0 +1,315 @@
+            return localSet.hgetall();
+          }));
+
+      Collection<ByteArrayWrapper> server2LocalHash =
+          server2.invoke(() -> (Collection<ByteArrayWrapper>) getLocalData(key, r -> {
+            RedisHash localSet = (RedisHash) r.get(new ByteArrayWrapper(key.getBytes()));
+            return localSet.hgetall();
+          }));
+
+      assertThat(server1LocalHash).containsExactlyInAnyOrder(server2LocalHash.toArray(
+          new ByteArrayWrapper[] {}));
+    }
+  }
+
+  @Test
+  public void shouldCorrectlyPropagateDeltaToSecondaryServer_whenRemovingFromHash() {
+    String key = "key";
+
+    Map<String, String> testMap = makeHashMap(ITERATION_COUNT, "field-", "value-");
+    jedis1.hset(key, testMap);
+
+    for (String field : testMap.keySet()) {
+      jedis1.hdel(key, field, testMap.get(field));
+
+      Collection<ByteArrayWrapper> server1LocalHash =
+          server1.invoke(() -> (Collection<ByteArrayWrapper>) getLocalData(key, r -> {
+            RedisHash localSet = (RedisHash) r.get(new ByteArrayWrapper(key.getBytes()));
+            if (localSet == null) {
+              return null;
+            }
+            return localSet.hgetall();
+          }));
+
+      Collection<ByteArrayWrapper> server2LocalHash =
+          server2.invoke(() -> (Collection<ByteArrayWrapper>) getLocalData(key, r -> {
+            RedisHash localSet = (RedisHash) r.get(new ByteArrayWrapper(key.getBytes()));
+            if (localSet == null) {
+              return null;
+            }
+            return localSet.hgetall();
+          }));
+
+      if (server1LocalHash == null || server2LocalHash == null) {
+        assertThat(server1LocalHash).isEqualTo(server2LocalHash);
+      } else {
+        assertThat(server1LocalHash).containsExactlyInAnyOrder(server2LocalHash.toArray(
+            new ByteArrayWrapper[] {}));
+      }
+    }
+  }
+
+  @Test
+  public void shouldCorrectlyPropagateDeltaToSecondaryServer_whenExpiring() {
+    String baseKey = "key-";
+
+    for (int i = 0; i < ITERATION_COUNT; i++) {
+      String key = baseKey + i;
+      jedis1.set(key, "value");
+      jedis1.expire(key, 20);
+      long server1LocalExpirtionTimestamp = server1.invoke(() -> (long) getLocalData(key, r -> {
+        RedisData localSet = r.get(new ByteArrayWrapper(key.getBytes()));
+        if (localSet == null) {
+          return null;
+        }
+        return localSet.getExpirationTimestamp();
+      }));
+
+      long server2LocalExpirationTimestamp = server2.invoke(() -> (long) getLocalData(key, r -> {
+        RedisData localSet = r.get(new ByteArrayWrapper(key.getBytes()));
+        if (localSet == null) {
+          return null;
+        }
+        return localSet.getExpirationTimestamp();
+      }));
+
+      assertThat(server1LocalExpirtionTimestamp).isEqualTo(server2LocalExpirationTimestamp);
+    }
+  }
+
+  private static Object getLocalData(String key,
```

Comment:
`key` is unused since it is captured by the lambdas and not used anywhere else other than the function being passed in. Don't worry about changing it unless there is something else you happen to also change before we merge in.